stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,850 @@
+        ),
+        axis=0,
+    )
+
+
+def _fred(t, mean_params):
+    """A fast rise exponential decay (FRED) flare shape.
+
+    Parameters
+    ----------
+    t:  jnp.ndarray
+        The time coordinates.
+    A:  jnp.int
+        Amplitude of the flare.
+    t0:
+        The location of the maximum.
+    phi:
+        Symmetry parameter of the flare.
+    delta:
+        Offset parameter of the flare.
+
+    Returns
+    -------
+    The y values for exponential flare.
+    """
+    A = jnp.atleast_1d(mean_params["A"])[:, jnp.newaxis]
+    t0 = jnp.atleast_1d(mean_params["t0"])[:, jnp.newaxis]
+    phi = jnp.atleast_1d(mean_params["phi"])[:, jnp.newaxis]
+    delta = jnp.atleast_1d(mean_params["delta"])[:, jnp.newaxis]
+
+    return jnp.sum(
+        A * jnp.exp(-phi * ((t + delta) / t0 + t0 / (t + delta))) * jnp.exp(2 * phi), axis=0
+    )
+
+
+def _get_kernel_params(kernel_type):
+    """
+    Generates a list of the parameters for the kernel for the GP model based on the kernel type.
+
+    Parameters
+    ----------
+    kernel_type: string
+        The type of kernel to be used for the Gaussian Process model
+        The parameters in log scale have a prefix of "log_"
+
+    Returns
+    -------
+        A list of the parameters for the kernel for the GP model
+    """
+    if kernel_type == "RN":
+        return ["log_arn", "log_crn"]
+    elif kernel_type == "QPO_plus_RN":
+        return ["log_arn", "log_crn", "log_aqpo", "log_cqpo", "log_freq"]
+    elif kernel_type == "QPO":
+        return ["log_aqpo", "log_cqpo", "log_freq"]
+    else:
+        raise ValueError("Kernel type not implemented")
+
+
+def _get_mean_params(mean_type):
+    """
+    Generates a list of the parameters for the mean for the GP model based on the mean type.
+
+    Parameters
+    ----------
+    mean_type: string
+        The type of mean to be used for the Gaussian Process model
+        The parameters in log scale have a prefix of "log_"
+
+    Returns
+    -------
+        A list of the parameters for the mean for the GP model
+    """
+    if (mean_type == "gaussian") or (mean_type == "exponential"):
+        return ["log_A", "t0", "log_sig"]
+    elif mean_type == "constant":
+        return ["log_A"]
+    elif (mean_type == "skew_gaussian") or (mean_type == "skew_exponential"):
+        return ["log_A", "t0", "log_sig1", "log_sig2"]
+    elif mean_type == "fred":
+        return ["log_A", "t0", "delta", "phi"]
+    else:
+        raise ValueError("Mean type not implemented")
+
+
+def get_gp_params(kernel_type, mean_type):
+    """
+    Generates a list of the parameters for the GP model based on the kernel and mean type.
+    To be used to set the order of the parameters for `get_prior` and `get_likelihood` functions.
+
+    Parameters
+    ----------
+    kernel_type: string
+        The type of kernel to be used for the Gaussian Process model
+
+    mean_type: string
+        The type of mean to be used for the Gaussian Process model
+
```

Comment:
Here, too, I think it would be helpful to list all the possible options for `kernel_type` and `mean_type` if this function is exposed to the user